### ISSUE TITLE
useStyletron returns a hook

### ISSUE THREAD

**Issue by conradreuter** [styletron/styletron#328]
The fact that the `css` function returned by `useStyletron` is in itself a hook is not really well documented. Maybe the documentation should at least mention that fact. Even better, call the function `useCss` to apply the ESLint rules of hooks.

**Comment by chasestarr** [comment]
Good idea - did the lack of documentation lead to unexpected behavior? Do you mind adding an example of that?

**Comment by ganemone** [comment]
+1 - all hooks should follow the `useX` pattern, otherwise the consumer won't know if they need to follow the rules of hooks. Similarly, invalid usage of the hook won't be caught by linters.

In this case, I think it would be interesting to see if there are any negative performance effects of squashing `useStyletron` into `useCss`. This would just entail moving the `useContext` calls at the top. So it would look like this

```js
// usage
import {useCss} from 'styletron-react';

function MyComponent() {
  const classA = useCss({...});
  const classB = useCss({...});
  return <div className={useCss({...})}>hello world</div>;
}

// implementation
export function useCss(style: styleObject) {
  const styletronEngine: StandardEngine = React.useContext(StyletronContext);
  const debugEngine = React.useContext(DebugEngineContext);
  const hydrating = React.useContext(HydrationContext);
  checkNoopEngine(styletronEngine);
  const className = driver(style, styletronEngine);
    if (!(__BROWSER__ && __DEV__)) {
      return className;
    }
    const {stack, message} = new Error("stacktrace source");
    const debugClassName = React.useMemo(
      () => {
        if (debugEngine && !hydrating) {
          return debugEngine.debug({
            stackInfo: {stack, message},
            stackIndex: 1,
          });
        }
        return "";
      },
      [debugEngine, hydrating],
    );
    return debugClassName ? `${debugClassName} ${className}` : className;
}

```

**Comment by rtsao** [comment]
@ganemone Ooh, that looks nice. I think that should be fine; I doubt `useContext` is expensive.

Maybe `import {useStyle} from "styletron-react"`

**Comment by dmaskasky** [comment]
@rtsao 
I think from a developers perspective, it wouldn't be obvious to give the return value of useStyle to the base component's className property.  Given the name useStyle, a developer might mistakenly do this:

``` js
import {useStyle} from "styletron-react";

const MyComponent = props => {
  const styles = useStyle({ color: 'blue' });
  return (
    <div style={styles}>Hello</div>
  );
}
```

**Comment by tajo** [comment]
We can just rename `css` to `useCss` in our docs?

@ganemone Could the implementation just be

```js
export function useCss(style: styleObject) {
  const [css] = useStyletron();
  return css(style);
}
```

One thing to consider, we expect that most developers will extend `useStyletron` to also provide a theme and use it like this (that's what we do in `baseui`):

```js
const [css, theme] = useStyletron();
```

Although, there is no need to couple styletron and theme, it seems practical since you almost always want to use those two at the same time.

**Comment by ganemone** [comment]
> ```js
> export function useCss(style: styleObject) {
>   const [css] = useStyletron();
>   return css(style);
> }
> ```

that works if you are adding a new hook, rather than replacing the existing one.

**Comment by kirbydcool** [comment]
Just ran into an issue with this. The docs make it look like `css` is a regular function and that you can use it inline.
```
<div className={css({ color: 'red' })} />
```
However, this will cause react to crash if you conditionally render an element because you're conditionally calling `css` which internally uses `useMemo`. This changes the number of hooks that are called between renders which is not allowed by react.

@rtsao 
It think the inlined syntax is ideal, but because of the hook issue it's not guaranteed to work and could cause hard to track down bugs. It looks like `css` only uses `useMemo` internally to optimize performance for the debug engine. https://github.com/styletron/styletron/blob/master/packages/styletron-react/src/index.js#L162
Would it make sense to remove that optimization to enable inline `css` (and make `css` not a hook)? Maybe there's another workaround?

**Comment by shannonmoeller** [comment]
I really like the `import {useCss}` idea. Would love to use it coupled with an `import {useTheme}`:

```js
export function useTheme() {
  const [, theme] = useStyletron();
  return theme;
}
```

**Comment by chasestarr** [comment]
Landed an update so that the returned function from `useStyletron` no longer contains a react hook. That function can now be called conditionally!

**Comment by chasestarr** [comment]
Closing this issue, published in `5.2.2`

**Comment by kirbydcool** [comment]
Thanks! Works great so far
